tests/test_app_capabilities.py

```
@@ -0,0 +1,74 @@
+    assert is_known_capability("network:https://example.com") is True
+    assert is_known_capability("network:garbage") is True
+
+
+def test_unknown_and_non_string_caps_are_rejected():
+    assert is_known_capability("app.bogus") is False
+    assert is_known_capability("files.read") is False  # not the shipped scheme
+    assert is_known_capability(123) is False
+
+
+def _manifest(perms):
+    return f"id: a\nname: A\nversion: 1.0.0\napp_type: web\npermissions:\n" + "".join(
```

Ruff (0.15.18)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)